ashley/src/main/kotlin/ktx/ashley/engines.kt

```diff
@@ -74,6 +75,22 @@ inline fun Engine.entity(configure: EngineEntity.() -> Unit = {}): Entity {
   return entity
 }
 
+/**
+ * Quick [EntitySystem] retrieval.
+ *
+ * @return the [EntitySystem] of the given type. May be null if it does not exist.
+ * @see Engine.getSystem
+ */
+inline fun <reified T : EntitySystem> Engine.getSystem(): T? = getSystem(T::class.java)
+
+/**
+ * Quick [EntitySystem] retrieval.
+ *
+ * @return the [EntitySystem] of the given type. May be null if it does not exist.
+ * @see Engine.getSystem
+ */
+inline operator fun <reified T : EntitySystem> Engine.get(type: KClass<T>): T? = getSystem(type.java)
```

Comment:
I don't think you need to inline this or use a reified type if you pass a `type` parameter. It will be a little bit more useful, since you can't dynamically call inlined methods with arbitrary types.

Reply:
I tried it without reified but then you lose the system type information and you just get a generic EntitySystem. But maybe I am missunderstanding something?

Reply:
This compiles for me, you might have confused it with the `getSystem` extension:

```kotlin
operator fun <T : EntitySystem> Engine.get(type: KClass<T>): T? = getSystem(type.java)
```
![image](https://user-images.githubusercontent.com/11707612/79596544-0b12cd80-80e1-11ea-9288-3733a786daa7.png)